RxCocoa/Common/DelegateProxy.swift

```diff
@@ -21,18 +21,20 @@ var dataSourceAssociatedTag: UnsafeRawPointer = UnsafeRawPointer(UnsafeMutablePo
 /// Base class for `DelegateProxyType` protocol.
 ///
 /// This implementation is not thread safe and can be used only from one thread (Main thread).
-open class DelegateProxy : _RXDelegateProxy {
-
+open class DelegateProxy<P: AnyObject, D: NSObjectProtocol>: _RXDelegateProxy, DelegateProxyBase {
```

Comment:
Why `DelegateProxyBase`. Maybe `DelegateProxyProtocol` is more idiomatic to Swift?